cartographer_ros/cartographer_ros/trajectory_options.h

```diff
@@ -35,6 +35,7 @@ struct TrajectoryOptions {
   bool provide_odom_frame;
   bool use_odometry;
   bool use_nav_sat;
+  bool project_pose_to_2d_plane;
```

Comment:
Do you think this needs to be a per-trajectory option? If not, move to node_options.h?

Reply:
Also, perhaps it would be better to just publish another transform instead, to a frame e.g. `tracking_frame + "_2d"`. This transform would be obtained by embedding the pose obtained from the pose extrapolator, which would still be published normally. No need to interfere with the operation of the pose extrapolator, IMHO.

Reply:
On the other hand, my proposal would make it more difficult to use the rest of the robot's TF tree with the additional `_2d` frame. I withdraw the previous comment :-)

Reply:
Yes, I also thought about just publishing an additional frame, but this would probably have unwanted side effects on the tree...

@per-trajectory option: I *personally* don't have a strong opinion on this tbh, however it sounds cleaner to me as a TrajectoryOption as the trajectory is projected into the 2D plane.

Reply:
That is true, but would a scenario in which one trajectory is projected, and one isn't make sense? If not, I vote for turning it into a node option.

Reply:
For me both scenarios are fine, let's wait for another opinion :)

Reply:
For consistency with the existing cartographer_ros options, TrajectoryOptions seems to fit better, even though we do not plan to change this option for each trajectory.